cldevice.go

```diff
@@ -38,6 +40,128 @@ const (
 
 var zeroSlice = []cl.CL_uint{cl.CL_uint(0)}
 
+func appendBitfield(info, value cl.CL_bitfield, name string, str *string) {
+	if (info & value) != 0 {
+		*str += name
+	}
+}
+
+func amdgpuFanPercentToValue(percent uint32) uint32 {
+	value := AMDGPUFanFailSafe
+
+	calculatedValue := float64(AMDGPUFanMax) * float64(percent) / float64(100)
+
+	if calculatedValue > 0 {
+		value = uint32(calculatedValue)
+	} else {
+		minrLog.Errorf("amdgpuFanPercentToValue() failed; using failsafe "+
+			"value of %v", AMDGPUFanFailSafe)
+	}
+
+	return value
+}
+
+// validate that we can write to the AMDGPU sysfs fan path
+func amdgpuFanPermissionsValid(index int) error {
+	path := amdgpuGetSysfsPath(index, "fan")
+
+	err := unix.Access(path, unix.W_OK)
+	if err != nil {
+		return fmt.Errorf("path %v is not writable", path)
+	}
+
+	return nil
+}
+
+func amdgpuGetSysfsPath(index int, field string) string {
+	cardPath := "/sys/class/drm/card" + strconv.Itoa(index)
```

Comment:
I explained this right before you commented.  AMDGPU is Linux only and there is no other way.